include/deal.II/lac/vector.templates.h

```diff
@@ -120,23 +120,32 @@ namespace internal
 
   template <typename Functor>
   void vectorized_transform(Functor &functor,
-                            size_type vec_size)
+                            size_type vec_size,
+                            std_cxx11::shared_ptr<parallel::internal::TBBPartitioner> &partitioner)
   {
 #ifdef DEAL_II_WITH_THREADS
     // only go to the parallel function in case there are at least 4 parallel
     // items, otherwise the overhead is too large
     if (vec_size >= 4*internal::Vector::minimum_parallel_grain_size &&
         MultithreadInfo::n_threads() > 1)
       {
+        if (partitioner.get() == NULL)
+          partitioner.reset(new parallel::internal::TBBPartitioner());
```

Comment:
I found the problematic part, see commit 2cae6f5. I now use an assertion at the place where we use the partitioner.